sphinx_gallery/tests/test_full.py

```diff
@@ -1278,6 +1279,23 @@ def test_recommend_n_examples(sphinx_app):
     assert count == n_examples
 
 
+def test_example_recommender_methods():
+    """Test that exactly n_examples thumbnails are displayed."""
```

Comment:
Is this docstring right? Also could we get the unit tests in their own file (so we don't have to run tiny build to run the test)? It's okay if we have to add a new test file e.g., `test_recommender`